web/controllers/api/release_controller.ex

```diff
@@ -37,41 +39,51 @@ defmodule HexWeb.API.ReleaseController do
   end
 
   def delete(conn, _params) do
+    package = conn.assigns.package
     release = conn.assigns.release
+
     multi =
       Ecto.Multi.new
       |> Ecto.Multi.delete(:release, Release.delete(release))
-      |> Ecto.Multi.insert(:log, audit(conn, "release.revert", {release.package, release}))
-
-    case HexWeb.Repo.transaction(multi) do
-      {:ok, _} ->
-        # TODO: Remove package from database if this was the only release
-        revert(release)
+      |> Ecto.Multi.insert(:log, audit(conn, "release.revert", {package, release}))
+
+    {:ok, conn} = HexWeb.Repo.transaction(fn ->
+      case HexWeb.Repo.transaction(multi) do
+        {:ok, _} ->
+          if Repo.aggregate(assoc(package, :releases), :count, :id) == 1 do
+            HexWeb.Repo.delete!(package)
```

Comment:
Any way we can avoid this race condition with locks or transaction isolation levels?